model/span.go

```diff
@@ -49,19 +49,19 @@ func (s *Span) Hash(w io.Writer) (err error) {
 }
 
 // HasSpanKind returns true if the span has a `span.kind` tag set to `kind`.
-func (s *Span) HasSpanKind(kind ext.SpanKindEnum) bool {
-	if tag, ok := KeyValues(s.Tags).FindByKey(string(ext.SpanKind)); ok {
-		return tag.AsString() == string(kind)
+func (s *Span) HasSpanKind(kind trace.SpanKind) bool {
+	if tag, ok := KeyValues(s.Tags).FindByKey("span.kind"); ok {
+		return tag.AsString() == kind.String()
 	}
 	return false
 }
 
 // GetSpanKind returns value of `span.kind` tag and whether the tag can be found
 func (s *Span) GetSpanKind() (spanKind string, found bool) {
-	if tag, ok := KeyValues(s.Tags).FindByKey(string(ext.SpanKind)); ok {
+	if tag, ok := KeyValues(s.Tags).FindByKey("span.kind"); ok {
 		return tag.AsString(), true
 	}
-	return "", false
+	return trace.SpanKindUnspecified.String(), false
```

Comment:
could we return the enum `trace.SpanKind` here instead of string? I don't suppose too many places in the code check for this.

Reply:
I didn't find that. I think enum is not supported for otel tracer span.

Reply:
@afzal442 [trace.SpanKind is an enum](https://github.com/open-telemetry/opentelemetry-go/blob/trace/v1.16.0/trace/trace.go#L428).

I like the idea of returning `trace.SpanKind` so it's more strongly and explicitly typed, but there are two drawbacks I see:
1. We'd need to map from `string` to `trace.SpanKind`. This is trivial.
2. If we want to use this type deep in the call stack, the changes may not be trivial and would need to be well tested, but doable as a single separate PR.

We could go half-way and return `trace.SpanKind` then call `spanKind.String()` wherever it's used but I don't think there's much value there.

Overall, my vote is for 1. as it's the most pragmatic.